packages/frontend/src/features/sqlRunner/hooks/useVirtualView.tsx

```diff
@@ -112,3 +114,45 @@ export const useUpdateVirtualView = (projectUuid: string) => {
         },
     });
 };
+
+const deleteVirtualView = async ({
+    projectUuid,
+    name,
+}: {
+    projectUuid: string;
+    name: string;
+}) =>
+    lightdashApi<ApiSuccessEmpty>({
+        url: `/projects/${projectUuid}/sqlRunner/virtual-view/${name}`,
+        method: 'DELETE',
+        body: undefined,
+    });
+
+export const useDeleteVirtualView = (projectUuid: string) => {
+    const history = useHistory();
+    const queryClient = useQueryClient();
+    const { showToastSuccess, showToastError } = useToaster();
+    return useMutation<
+        ApiSuccessEmpty,
+        ApiError,
+        { projectUuid: string; name: string }
+    >({
+        mutationFn: deleteVirtualView,
+        onSuccess: async () => {
+            await queryClient.invalidateQueries({
+                queryKey: ['tables', projectUuid, 'filtered'],
+            });
+
+            showToastSuccess({
+                title: 'Success! Virtual view deleted',
+            });
+
+            history.push(`/projects/${projectUuid}/tables`);
+        },
+        onError: () => {
+            showToastError({
+                title: 'Error deleting virtual view',
```

Comment:
no description about the error ?

Reply:
I think it's fine since it only errors if it's db issue or it's a permissions issue. Actually, this makes me think we need to hide the edit and delete option for people who can't manage explores. I'm going to create a ticket for this